### PR TITLE
Url proxy site for maintaining consistent friendly urls

### PR DESCRIPTION
**Pull request by ouchadam**
- A firebase function which reads the request path to match a key against a url in a firebase realtime database to redirect to.

- A react website to admin (add/remove) links

screen shot 2018-05-01 at 10 14 50

### CODE REVIEW

File: links/site/src/index.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+}
+
+const AuthComponent = ({ }) => {
+    return (
+        <button onClick={onSignInClicked()}> Sign in</button>
+    )
+}
+
+const onSignInClicked = () => () => {
```
Comment (takecare) [inline links/site/src/index.js]: why `() => () =>`?

File: links/site/src/index.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    componentDidMount() {
+        this.unsubscribe = firebase.auth().onAuthStateChanged(user => {
+            this.setState({
+                isAuthed: Boolean(user) && !user.isAnonymous
+            })
+        })
+    }
+
+    componentWillUnmount() {
+        this.unsubscribe()
```
Comment (takecare) [inline links/site/src/index.js]: does it matter at this point to change the state?

Comment (ouchadam) [reply links/site/src/index.js]: react warns about memory leaks due to an async which may not have finished when unmounting

File: links/site/src/index.js (hunk 1 above)
Comment (ouchadam) [reply links/site/src/index.js]:
```
        <button onClick={onSignInClicked()}> Sign in</button>
```

lol no idea, will remove the unneeded layer

**Pull request merged**